packages/core/src/util/renderer.ts

```diff
@@ -226,10 +241,15 @@ const extractRequired = (
   let currentSchema = schema;
   while (
     i < prevSegments.length &&
-    has(currentSchema, 'properties') &&
-    has(get(currentSchema, 'properties'), prevSegments[i])
+    (has(currentSchema, prevSegments[i]) ||
+      (has(currentSchema, 'properties') &&
+        has(get(currentSchema, 'properties'), prevSegments[i])))
```

Comment:
Since now we are adding `/items` in the schema path, then maybe there should be another condition for it. For example would a path like this cause a problem: `#/properties/object/properties/array/items/properties/nestedArray/items/properties/propertyName`?